### PR TITLE
[RFR] Create Ansible service dialog through REST API

### PR DESCRIPTION
**Pull request by nachandr**
1)Created Ansible service dialog through REST API
2)The existing tests through which Tower catalog items are ordered now use the new service dialog
3)Also added a new test to verify that Ansible jobs run on limited hosts.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit ffae466f9c5d187c14218541686878b2d694b5a8:

`cfme/tests/services/test_config_provider_servicecatalogs.py`:
- :bangbang: Line 67:1: **E305** *expected 2 blank lines after class or function definition, found 1*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit aa7aaf4f4cf0fae2a80d150f7f7d7f401b65fd4e:

`cfme/tests/services/test_config_provider_servicecatalogs.py`:
- :bangbang: Line 67:1: **E305** *expected 2 blank lines after class or function definition, found 1*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/fixtures/ansible_tower.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+@pytest.fixture(scope="function")
+def ansible_tower_dialog(request, appliance):
+    """Returns service dialog object."""
+    rest_resource = ansible_tower_dialog_rest(request, appliance)
+    service_dialogs = appliance.collections.service_dialogs
+    service_dialog = service_dialogs.instantiate(
+        label=rest_resource.label,
+        description=rest_resource.description)
+    return service_dialog
```
Comment (sshveta) [inline cfme/fixtures/ansible_tower.py]: Can you use yield intead and Clean up dialog after yield.

Comment (nachandr) [reply cfme/fixtures/ansible_tower.py]: Done

**Comment by dajoRH** [comment]
Lint report for commit 5ff31f3393b6c981a595eab15cf6c00df5b08d93:

`cfme/fixtures/ansible_tower.py`:
- :bangbang: Line 124:8: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/fixtures/ansible_tower.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import pytest
+
+from cfme.rest.gen_data import _creating_skeleton
+from cfme.utils.blockers import GH
+
+
+def ansible_tower_dialog_rest(request, appliance):
+    """Creates service dialog using REST API."""
+    uid = fauxfactory.gen_alphanumeric()
+    data = {
```
Comment (john-dupuy) [inline cfme/fixtures/ansible_tower.py]: @nachandr Thoughts on putting this in a yaml file in `integration_tests.data` directory? That way this file will not be overcrowded with this dictionary.

File: cfme/fixtures/ansible_tower.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    """Returns service dialog object."""
+    rest_resource = ansible_tower_dialog_rest(request, appliance)
+    service_dialogs = appliance.collections.service_dialogs
+    service_dialog = service_dialogs.instantiate(
+        label=rest_resource.label,
+        description=rest_resource.description)
+    yield service_dialog
+
+    if not GH(8836).blocks or appliance.version != '5.11':
+        if service_dialog.exists:
```
Comment (john-dupuy) [inline cfme/fixtures/ansible_tower.py]: Q: Can you use `delete_if_exists` here?

File: cfme/tests/services/test_config_provider_servicecatalogs.py [hunk 4]
```diff
@@ -75,7 +78,15 @@ def test_order_tower_catalog_item(appliance, catalog_item, request, job_type):
         casecomponent: Services
         caseimportance: high
     """
-    service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
+    if job_type == 'template_limit':
+        config_manager_obj = config_manager
```
Comment (john-dupuy) [inline cfme/tests/services/test_config_provider_servicecatalogs.py]: Q: Why not just use `config_manager` in L83 below? I.e. why is this variable assignment necessary?

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/fixtures/ansible_tower.py (hunk 3 above)
Comment (nachandr) [reply cfme/fixtures/ansible_tower.py]: Done

File: cfme/tests/services/test_config_provider_servicecatalogs.py (hunk 4 above)
Comment (nachandr) [reply cfme/tests/services/test_config_provider_servicecatalogs.py]: Done

**Comment by dajoRH** [comment]
Lint report for commit 21338610e95388b6a6ff6a6c7dd5971186dcc067:

`cfme/services/catalogs/catalog_items/__init__.py`:
- :bangbang: Line 345:12: **E999** *IndentationError: unexpected indent*
- :bangbang: Line 345:13: **E113** *unexpected indentation*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Pull request merged**